platform/mellanox/mlnx-ssd-fw-update.sh

```diff
@@ -683,7 +690,12 @@ elif [ $ARG_UPDATE_FLAG == $TRUE ]; then
                 else
                     LOG_MSG "SSD FW update completed successfully."
 
-                    if [[ "yes" == "$power_policy" || $ARG_POWER_CYCLE_FLAG == $TRUE ]]; then
+		    if [[ "yes" == "$power_policy" || $ARG_POWER_CYCLE_FLAG == $TRUE ]]; then
+
+                        if [[ $ARG_FORCE_NO_POWER_CYCLE_FLAG == $TRUE ]]; then
+				LOG_MSG_AND_EXIT "An IMMEDIATE power cycle is REQUIRED to upgrade the SSD. Please perform a cold reboot as soon as possible."
```

Comment:
@alexrallen please speak with BSP on how to handle it properly for `Innodisk`. We do need to have a protection mechanism in case the user will invoke it manually with `--no-power-cycle`

Reply:
Mentioned this in the thread with BSP.